frontend/public/components/utils/resource-icon.tsx

```diff
@@ -2,22 +2,20 @@ import * as React from 'react';
 import * as classNames from 'classnames';
 import * as _ from 'lodash-es';
 
-import { K8sResourceKindReference } from '../../module/k8s';
-import { modelFor } from '../../module/k8s/k8s-models';
-import { kindToAbbr } from '../../module/k8s/get-resources';
+import { K8sKind, K8sResourceKindReference, modelFor, kindToAbbr } from '../../module/k8s';
 
 const MEMO = {};
 
-export const ResourceIcon: React.SFC<ResourceIconProps> = ({ className, kind }) => {
+export const ResourceIcon: React.SFC<ResourceIconProps> = ({ className, kind, model }) => {
   // if no kind, return null so an empty icon isn't rendered
-  if (!kind) {
+  if (!kind && !model) {
     return null;
   }
   const memoKey = className ? `${kind}/${className}` : kind;
   if (MEMO[memoKey]) {
     return MEMO[memoKey];
   }
```

Comment:
I think this might be a problem as well... as it will key off of the `kind` and potentially reuse the wrong model rendering.